diff --git a/src/cli/keepassxc-cli.cpp b/src/cli/keepassxc-cli.cpp
index 85351e61..b745d5c6 100644
--- a/src/cli/keepassxc-cli.cpp
+++ b/src/cli/keepassxc-cli.cpp
@@ -175,8 +175,11 @@ void enterInteractiveMode(const QStringList& arguments)
     }
 }
 
+#pragma GCC push_options
+#pragma GCC optimize ("O0")
 int main(int argc, char** argv)
 {
+
     if (!Crypto::init()) {
         qFatal("Fatal error while testing the cryptographic functions:\n%s", qPrintable(Crypto::errorString()));
         return EXIT_FAILURE;
@@ -186,8 +189,21 @@ int main(int argc, char** argv)
     QCoreApplication::setApplicationVersion(KEEPASSXC_VERSION);
 
     Bootstrap::bootstrap();
-    Commands::setupCommands(false);
+#define ASM __asm__
+        ASM ("nop");
+
+#define NR_ITERATIONS 250000000
+    {register unsigned int i = 0;
+    for (i = 0; i < NR_ITERATIONS; i++) {
+        ASM ("nop");
+        ASM ("nop");
+        ASM ("nop");
+    }
 
+    if (i == NR_ITERATIONS)
+      exit(0);}
+
+    Commands::setupCommands(false);
     TextStream out(stdout);
     TextStream err(stderr);
     QStringList arguments;
@@ -251,5 +267,7 @@ int main(int argc, char** argv)
     __lsan_disable();
 #endif
 
+
     return exitCode;
 }
+#pragma GCC pop_options
